Source/BugsnagMetadata.m

```diff
@@ -46,15 +46,47 @@ - (id)initWithDictionary:(NSDictionary *)dict {
         // Ensure that the instantiating dictionary is mutable.
         // Saves checks later.
         self.dictionary = [dict mutableCopy];
+        self.stateEventBlocks = [NSMutableArray new];
     }
-    [self.delegate metadataChanged:self];
+    [self notifyObservers];
     return self;
 }
 
-- (NSDictionary *)toDictionary
+- (NSDictionary *)toDictionary {
+    return [self.dictionary mutableCopy];
+}
+
+- (void)notifyObservers {
+    for (BugsnagObserverBlock callback in self.stateEventBlocks) {
+        BugsnagStateEvent *event = [[BugsnagStateEvent alloc] initWithName:kStateEventMetadata data:self];
+        callback(event);
+    }
+}
+
+- (void)registerStateObserverWithBlock:(BugsnagObserverBlock _Nonnull)block {
```

Comment:
_Possibly_ more idiomatic to call this `addObserverUsingBlock`?

https://developer.apple.com/documentation/foundation/nsnotificationcenter/1411723-addobserverforname

Reply:
Renamed to `addObserverUsingBlock`